pkg/assembler/backends/ent/backend/cve.go

```diff
@@ -0,0 +1,148 @@
+package backend
+
+import (
+	"context"
+	"strings"
+
+	"entgo.io/ent/dialect/sql"
+	"github.com/guacsec/guac/pkg/assembler/backends/ent"
+	"github.com/guacsec/guac/pkg/assembler/backends/ent/securityadvisory"
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+)
+
+type advisoryQuerySpec struct {
+	CveID  *string
+	GhsaID *string
+	OsvID  *string
+	Year   *int
+	ID     *string
+}
+
+func (b *EntBackend) Cve(ctx context.Context, spec *model.CVESpec) ([]*model.Cve, error) {
+	results, err := getAdvisories(ctx, b.client, &advisoryQuerySpec{
+		ID:    spec.ID,
+		CveID: spec.CveID,
+		Year:  spec.Year,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return collect(results, toModelCVE), nil
+}
+
+func (b *EntBackend) IngestCve(ctx context.Context, spec *model.CVEInputSpec) (*model.Cve, error) {
```

Comment:
I think you might want it like this instead, otherwise client is not passed correctly, and then remove the `WithinTX` from `upsertAdvisory`

```
func (b *EntBackend) IngestCve(ctx context.Context, spec *model.CVEInputSpec) (*model.Cve, error) {
	advisory, err := WithinTX(ctx, b.client, func(ctx context.Context) (*ent.SecurityAdvisory, error) {
		client := ent.FromContext(ctx)
		return upsertAdvisory(ctx, client, advisoryQuerySpec{
			CveID: &spec.CveID,
			Year:  &spec.Year,
		})
	})
	if err != nil {
		return nil, err
	}
	return toModelCVE(advisory), nil
}
```

Reply:
I guess for consistency that makes sense... `upsertAdvisory` simply wraps this logic for CVEs, GHSAs, and OSV since they are stored in the same table. At the moment this is the only callsite for upserts, but if we end up needing it to not be transactional I think that change would make sense.